rsconnect/bundle.py

```diff
@@ -50,6 +51,118 @@
 ]
 directories_to_ignore = {Path(d) for d in directories_ignore_list}
 
+mimetypes.add_type("text/ipynb", ".ipynb")
+
+
+class Manifest:
```

Comment:
I like the new `Manifest` and `Bundle` classes. Let's follow up after this PR and use them for the other deployment types.